web3/version.py

```diff
@@ -44,8 +44,7 @@ def ethereum(self):
 class Version(Module):
     @property
     def api(self):
-        from web3 import __version__
-        return __version__
+        raise DeprecationWarning("This method has been deprecated as of EIP 1474.")
```

Comment:
I know there was a comment about raising warnings vs not in a past PR, but why are we using one over the other? Also, maybe this warning should say something like: `This method has been deprecated ... Please use web3.api instead`?

Reply:
Good call on the warning msg! re other stuff: from [here](https://github.com/ethereum/web3.py/pull/1271#discussion_r264277304)

> So warnings don't break codeflow, they just dump information out so a user who's watching for them knows something is going away. Using the raise syntax breaks code flow so they are two separate mechanisms. These can be easy to mixup since the exception class is called `DeprecationWarning` but they are two different mechanisms that use the same class.

> warning.warn(...) - for v4
> raise DeprecationWarning - for v5